core/server/data/migrations/versions/3.12/01-remove-legacy-is-paid-flag-from-settings.js

```diff
@@ -0,0 +1,110 @@
+const common = require('../../../../lib/common');
+const debug = require('ghost-ignition').debug('migrations');
+
+module.exports.config = {
+    transaction: true
+};
+
+module.exports.up = (options) => {
+    let localOptions = _.merge({
+        context: {internal: true}
+    }, options);
+    const settingsKey = 'members_subscription_settings';
+
+    return localOptions
+        .transacting('settings')
+        .then((response) => {
+            if (!response) {
+                common.logging.warn('Cannot find settings.');
+                return;
+            }
+
+            let subscriptionSettingsEntry = response.find((entry) => {
+                return entry.key === settingsKey;
+            });
+
+            if (!subscriptionSettingsEntry) {
+                common.logging.warn('Cannot find members subscription settings.');
+                return;
+            }
+
+            let subscriptionSettings = JSON.parse(subscriptionSettingsEntry.value);
+
+            debug('before cleanup');
+            debug(JSON.stringify(subscriptionSettings, null, 2));
+
+            const hasIsPaid = Object.prototype.hasOwnProperty.call(subscriptionSettings, 'isPaid');
+
+            if (hasIsPaid) {
+                debug('Removing legacy isPaid flag from members settings');
+                delete subscriptionSettings.isPaid;
+            }
+
+            debug('after cleanup');
+            debug(JSON.stringify(subscriptionSettings, null, 2));
+
+            return localOptions
+                .transacting('settings')
+                .where('key', settingsKey)
+                .update({
+                    value: JSON.stringify(subscriptionSettings)
+                });
+        });
+};
+
+module.exports.down = (options) => {
+    let localOptions = _.merge({
+        context: {internal: true}
+    }, options);
+    const settingsKey = 'members_subscription_settings';
+
+    return localOptions
+        .transacting('settings')
+        .then((response) => {
+            if (!response) {
+                common.logging.warn('Cannot find settings.');
+                return;
+            }
+
+            let subscriptionSettingsEntry = response.find((entry) => {
+                return entry.key === settingsKey;
+            });
+
+            if (!subscriptionSettingsEntry) {
+                common.logging.warn('Cannot find members subscription settings.');
+                return;
+            }
+
+            let subscriptionSettings = JSON.parse(subscriptionSettingsEntry.value);
+
+            debug('before cleanup');
+            debug(JSON.stringify(subscriptionSettings, null, 2));
+
+            let isPaid = false;
+
+            const stripePaymentProcessor = subscriptionSettings.paymentProcessors.find(
+                paymentProcessor => paymentProcessor.adapter === 'stripe'
+            );
+
+            if (stripePaymentProcessor && stripePaymentProcessor.config.public_token && stripePaymentProcessor.config.secret_token) {
```

Comment:
@allouis does this logic make sense to you? Couldn't find any concrete reference to equivalent of this flag

Reply:
Might wanna doublecheck with @rishabhgrg who introduced the flag, too :relaxed:

Reply:
@gargol Think this is fine, `isPaid` was to mark if stripe setup is present or not so this should ideally cover it. 🤔